crates/router/src/core/admin.rs

```diff
@@ -3477,6 +3481,176 @@ pub async fn update_business_profile(
             .await?,
     ))
 }
+#[cfg(all(feature = "v2", feature = "business_profile_v2"))]
+pub async fn update_business_profile(
+    state: SessionState,
+    profile_id: &str,
+    merchant_id: &id_type::MerchantId,
+    request: api::BusinessProfileUpdate,
+) -> RouterResponse<api::BusinessProfileResponse> {
+    let db = state.store.as_ref();
+    let business_profile = db
+        .find_business_profile_by_profile_id(profile_id)
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::BusinessProfileNotFound {
+            id: profile_id.to_owned(),
+        })?;
+    let key_store = db
```

Comment:
Merchant Key store is not required for v2 instead we should use the secret manager approach